lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -953,11 +941,14 @@ def _relation_hosts(self, relation) -> dict:
             A dictionary that maps unit names to unit addresses for
             the specified relation.
         """
-        return {
-            unit.name: relation.data[unit].get("prometheus_scrape_host")
-            for unit in relation.units
-            if relation.data[unit].get("prometheus_scrape_host")
-        }
+        hosts = {}
+        for unit in relation.units:
+            unit_name = relation.data[unit].get("prometheus_scrape_unit_name")
+            host_address = relation.data[unit].get("prometheus_scrape_unit_address")
```

Comment:
If we do a fallback to the previous label name we do not break compat with the previous versions of this lib.
Also, the name of the variable is wrong :-)

Reply:
You mean leave the name as `prometheus_scrape_host` ?

Reply:
OK. I presume we do intend to deprecate `prometheus_scrape_host` at some point. I will put a TODO to this effect.

Reply:
Done.